speech/cloud-client/beta_snippets.py

```diff
@@ -156,33 +152,46 @@ def transcribe_file_with_diarization(speech_file):
 
     config = speech.types.RecognitionConfig(
         encoding=speech.enums.RecognitionConfig.AudioEncoding.LINEAR16,
-        sample_rate_hertz=16000,
+        sample_rate_hertz=8000,
         language_code='en-US',
         enable_speaker_diarization=True,
         diarization_speaker_count=2)
 
     print('Waiting for operation to complete...')
     response = client.recognize(config, audio)
 
-    for i, result in enumerate(response.results):
-        alternative = result.alternatives[0]
-        print('-' * 20)
-        print('First alternative of result {}: {}'
-              .format(i, alternative.transcript))
-        print('Speaker Tag for the first word: {}'
-              .format(alternative.words[0].speaker_tag))
+    # The transcript within each result is separate and sequential per result.
+    # However, the words list within an alternative (for whatever reason)
+    # includes all the words from all the results thus far. Thus, to get all
+    # the words with speaker tags, you only have to take the words list from
+    # the last result:
+    result = response.results[-1]
+
+    words_info = result.alternatives[0].words
+
+    # Separating the words by who said what:
+    speakers = []
+    words = []
+    for word_info in words_info:
+        if (not speakers) or speakers[-1] != word_info.speaker_tag:
```

Comment:
I think this implementation means that if speakers appear in certain orders you could end up with duplicates. For example.

`word_info` speakers: `A, B, B, A` would result in a `speakers` lists: `A, B, A`.

I think this sample would be clearer with a single dictionary of speakers to words rather than two lists.

Reply:
We had a long discussion about the code readability of this part of the sample. The API returns each word with an associated speak_tag. So, if I am speaker #1 and I say "I am ok", the words 'I' and 'am' and 'okay' will all be tagged with speaker_tag: 1.

Since the point of the sample is mostly just to showcase the API call and how to unbox the response object, I am beginning to think that this particular implementation is adding more confusion than clarity. I am leaning back toward a simpler implementation that just iterates over words and prints each word along with the associated speaker_tag. What do you think?

Reply:
Yes, printing each word with a speaker tag would be much clearer.

You could do something like
```
current_speaker = None
for word_info in words_info:
    if current_speaker is None or current_speaker != word_info.speaker_tag:
        current_speaker = word_info.speaker_tag
        print()
        print(current_speaker)
    print(word_info.word)
```
if you wanted to keep them grouped.